sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -668,24 +668,63 @@ ESIMD_INLINE ESIMD_NODEBUG
   __esimd_slm_write<T, n>(offsets.data(), vals.data(), pred.data());
 }
 
+/// Gathering read from the SLM given specified \p offsets.
+/// Up to 4 data elements may be accessed at each address depending on the
+/// enabled channel \p Mask.
+/// \tparam T element type of the returned vector. Must be 4-byte.
+/// \tparam N size of the \p offsets vector. Must be 8, 16 or 32.
+/// \tparam Mask represents a pixel's channel mask.
+/// @param offsets byte-offsets within the SLM.
+/// @param pred predication control used for masking lanes.
+/// \ingroup sycl_esimd
+template <typename T, int N, rgba_channel_mask Mask>
+ESIMD_INLINE ESIMD_NODEBUG typename sycl::detail::enable_if_t<
+    (N == 8 || N == 16 || N == 32) && (sizeof(T) == 4),
+    simd<T, N * get_num_channels_enabled(Mask)>>
+slm_load_rgba(simd<uint32_t, N> offsets, simd<uint16_t, N> pred = 1) {
```

Comment:
let's rename it to match non-SLM variant - `slm_gather_rgba`.
@rolandschulz - I recall that what we discussed.

Reply:
I agree. I can do renaming in a separate commit since slm_load/store also needs to be renamed.
- slm_load -> slm_gather
- slm_load_rgba -> slm_gather_rgba
- slm_store -> slm_scatter
- slm_store_rgba -> slm_scatter_rgba

@rolandschulz, do you agree?

Reply:
> I can do renaming in a separate commit since slm_load/store also needs to be renamed.

I don't see a point in doing this in a separate commit, frankly. First commit would be slm_load4->slm_load_rgba, second - slm_load_rgba->slm_gather_rgba. Why not slm_load4 -> slm_gather_rgba in one shot?

Reply:
Renamed. E2E test for slm_load/store will be provided in https://github.com/intel/llvm-test-suite/pull/380